osu.Game/Screens/SelectV2/SongSelect.cs

```diff
@@ -160,19 +174,22 @@ protected override void LogoExiting(OsuLogo logo)
             logo.FadeOut(120, Easing.Out);
         }
 
-        private partial class SoloModSelectOverlay : UserModSelectOverlay
+        /// <summary>
+        /// Set the query to the search text box.
+        /// </summary>
+        /// <param name="query">The string to search.</param>
+        public void Search(string query)
         {
-            protected override bool ShowPresets => true;
+            carousel.Filter(new FilterCriteria
+            {
+                // TODO: this should only set the text of the current criteria, not use a completely new criteria.
+                SearchText = query,
+            });
         }
 
-        private partial class PlayerLoaderV2 : PlayerLoader
+        private partial class SoloModSelectOverlay : UserModSelectOverlay
```

Comment:
Can we put this in `SoloSongSelect`? Quite strange for this to be here now that `SoloSongSelect` has been spawned.

Reply:
I'm mostly copying the existing song select which has this as a base default. Not yet sure how to handle that going forwards..

Maybe just rename for now to `DefaultModeSelectOverlay`? Dunno.

Reply:
It'll be a bit of a detour but you could make `ShowPresets` an init property or something instead of a virtual.

Reply:
Sure, done.